private/tools/java/rules/jvm/external/zip/StableZipEntry.java

```diff
@@ -17,15 +17,15 @@
 
 package rules.jvm.external.zip;
 
-import java.nio.file.attribute.FileTime;
-import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
 import java.util.zip.ZipEntry;
 
 public class StableZipEntry extends ZipEntry {
 
   // File time is taken from the epoch (1970-01-01T00:00:00Z), but zip files
   // have a different epoch. Oof. Make something sensible up.
-  private static final long DOS_EPOCH = Instant.parse("1985-02-01T00:00:00.00Z").toEpochMilli();
+  private static final LocalDateTime DOS_EPOCH = LocalDateTime.of(1985,2,1,0,0);
```

Comment:
The idea with this was that it was UTC "1st February, 1985". I _think_ the constant we're looking for here is `long DOS_EPOCH = 473472000000` Using local timezones and timestamps will lead to the build being different depending on where you are in the world, and the date, when you run the build

Reply:
I'm pretty sure that with the `ZipEntry` API we can't use a single timestamp, but need to get a timestamp that is adjusted with the system timezone so it gets "cancelled out" when we pass it into the `setTime()` method. The same approach is used in the Bazel repo here:

https://github.com/bazelbuild/bazel/blob/ef3f05852b3bf542903f091184da96b96354a13d/src/java_tools/buildjar/java/com/google/devtools/build/buildjar/jarhelper/JarHelper.java#L58-L62

and here:

https://github.com/bazelbuild/bazel/blob/ef3f05852b3bf542903f091184da96b96354a13d/src/tools/android/java/com/google/devtools/build/android/AarGeneratorAction.java#L73-L75

Reply:
Thank you for the links. I also played around with the APIs and went and read the source of what the zip output stream is doing. I regret some life decisions, but this looks good to me. Thank you for clarifying :)